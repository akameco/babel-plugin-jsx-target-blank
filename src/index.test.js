// @flow
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  tests: [
    {
      title: 'default',
      code: `
        <a href="akameco.github.io" target="_blank">akameco</a>
    `,
    },
    {
      title: 'href = variable',
      code: `
      <a href={externalLink} target="_blank">akameco</a>
    `,
    },
    {
      title: 'has secure rel',
      code: `
      <a href="akameco.github.io" target="_blank" rel="noopener noreferrer">
        akameco
      </a>
    `,
    },
    {
      title: 'has noopener only',
      code: `
      <a href="akameco.github.io" target="_blank" rel="noopener">
        akameco
      </a>
    `,
    },
    {
      title: 'with other rel',
      code: `
      <a href="akameco.github.io" target="_blank" rel="noopener prefetch">
        akameco
      </a>
    `,
    },
    {
      title: 'has no rel',
      code: `
      <a href="akameco.github.io" target="_blank" rel="">
        akameco
      </a>
    `,
    },
    {
      title: 'rel = variable',
      code: `
      <a href="akameco.github.io" target="_blank" rel={val}>
        akameco
      </a>
    `,
    },
    {
      title: 'no-a-tag',
      code: `<div id="hoge">hello</div>`,
    },
  ],
})
