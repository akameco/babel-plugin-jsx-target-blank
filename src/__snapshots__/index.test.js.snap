// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default default: default 1`] = `
"
<a href=\\"akameco.github.io\\" target=\\"_blank\\">akameco</a>

      ↓ ↓ ↓ ↓ ↓ ↓

<a href=\\"akameco.github.io\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">akameco</a>;
"
`;

exports[`default has no rel: has no rel 1`] = `
"
<a href=\\"akameco.github.io\\" target=\\"_blank\\" rel=\\"\\">
  akameco
</a>

      ↓ ↓ ↓ ↓ ↓ ↓

<a href=\\"akameco.github.io\\" target=\\"_blank\\" rel=\\" noopener noreferrer\\">
  akameco
</a>;
"
`;

exports[`default has noopener only: has noopener only 1`] = `
"
<a href=\\"akameco.github.io\\" target=\\"_blank\\" rel=\\"noopener\\">
  akameco
</a>

      ↓ ↓ ↓ ↓ ↓ ↓

<a href=\\"akameco.github.io\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
  akameco
</a>;
"
`;

exports[`default has secure rel: has secure rel 1`] = `
"
<a href=\\"akameco.github.io\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
  akameco
</a>

      ↓ ↓ ↓ ↓ ↓ ↓

<a href=\\"akameco.github.io\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
  akameco
</a>;
"
`;

exports[`default href = variable: href = variable 1`] = `
"
<a href={externalLink} target=\\"_blank\\">akameco</a>

      ↓ ↓ ↓ ↓ ↓ ↓

<a href={externalLink} target=\\"_blank\\" rel=\\"noopener noreferrer\\">akameco</a>;
"
`;

exports[`default no-a-tag: no-a-tag 1`] = `
"
<div id=\\"hoge\\">hello</div>

      ↓ ↓ ↓ ↓ ↓ ↓

<div id=\\"hoge\\">hello</div>;
"
`;

exports[`default rel = variable: rel = variable 1`] = `
"
<a href=\\"akameco.github.io\\" target=\\"_blank\\" rel={val}>
  akameco
</a>

      ↓ ↓ ↓ ↓ ↓ ↓

<a href=\\"akameco.github.io\\" target=\\"_blank\\" rel={val}>
  akameco
</a>;
"
`;

exports[`default with other rel: with other rel 1`] = `
"
<a href=\\"akameco.github.io\\" target=\\"_blank\\" rel=\\"noopener prefetch\\">
  akameco
</a>

      ↓ ↓ ↓ ↓ ↓ ↓

<a href=\\"akameco.github.io\\" target=\\"_blank\\" rel=\\"noopener prefetch noreferrer\\">
  akameco
</a>;
"
`;
